src/test/java/com/zombiecastlerush/building/ShopTest.java

```diff
@@ -0,0 +1,58 @@
+package com.zombiecastlerush.building;
+
+import com.zombiecastlerush.entity.Player;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class ShopTest {
+    Player p1;
+    Shop s1;
+    Item i1 = new Item("Magic-Lamp", "Genie's houe", 500.0);
+    Item i2 = new Item("Flying-Carpet", "Let's tour the sky", 200.0);
+
+
+    @Before
+    public void setUp() throws Exception {
+        p1 = new Player("testPlaye01");
+        s1 = new Shop("S1", "Shop1");
+        p1.setCurrentPosition(s1);
+        s1.getInventory().addItems(i1);
+        p1.getInventory().addItems(i2);
+    }
+
+    @Test
+    public void testBuyWhenPlayerDoesNotHaveEnoughMoney() {
+        //can't buy when player does not have enough money
+        double preBuy_balance = p1.getAcctBalance();
+        s1.buy(p1, i1);
+        assertFalse(p1.getInventory().getItems().contains(i1));
+        assertTrue(s1.getInventory().getItems().contains(i1));
+        assertEquals(preBuy_balance, p1.getAcctBalance(), 0.001);
```

Comment:
Would `assertEquals(0.001, p1.getAcctBalance())` not work here?

Reply:
it works, but to handle deprecated msg from IDE for assertEqual(double,double)